Homework5/SparseMatrix/SparseMatrix.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SparseMatrix
+{
+    public class SparseMatrix : IEnumerable<int>
+    {
+        public int Rows { get; }
+        public int Columns { get; }
+
+        private Dictionary<ElementDirections, int> arrayElements = new Dictionary<ElementDirections, int>();
+
+        public SparseMatrix(int columns, int rows)
+        {
+            if (rows <= 0 || columns <= 0)
+            {
+                throw new ArgumentException();
+            }
+            else
+            {
+                Rows = rows;
+                Columns = columns;
+            }
+
+        }
+
+        public int this[int column, int row]
+        {
+            get
+            {
+                if (row < 0 ||
+                    column < 0 ||
+                    row >= Rows ||
+                    column >= Columns)
+                {
+                    throw new IndexOutOfRangeException();
+                }
+
+                int itemToReturn = default;
+
+                foreach (var element in arrayElements)
+                {
+                    if (element.Key.Equals(new ElementDirections(column, row)))
+                    {
+                        itemToReturn = element.Value;
+                        break;
+                    }
+
+                }
+
+                return itemToReturn;
+            }
+            set
+            {
+                if (row < 0 ||
+                    column < 0 ||
+                    row >= Rows ||
+                    column >= Columns)
+                {
+                    throw new IndexOutOfRangeException();
+                }
+                else
+                {
+                    ElementDirections directions = new ElementDirections(column, row);
+
+                    if (arrayElements.ContainsKey(directions))
+                    {
+                        arrayElements[directions] = value;
```

Comment:
Do not store value when it is 0, ignore such values. 
Sparse matrix may have only few non-zero elements, lets keep only them to support good performance characteristics of the app.

Reply:
Requested changes above unresolved. 
**This setter should not store 0 values.** 
Assume we have a sparse matrix 1 mln x 1mln elements and it has single non-zero element.
Current code will take memory of size 10^12 bytes * N, where N is size of 1 element.

Reply:
I'm very sorry, I thought I've had it fixed,  as previously I've added another else if, right after checking rows and columns, and returning if value == 0.  However, now I've changed, that the value assignment would be proceeded only if the value != 0. I'm not sure if I'm putting the check in the right place, but it seems to be working.